src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,699 @@
+        );
+
+        /// optimistically transfer funds to the specified pcv deposit
+        _movePCV(
+            source,
+            destination,
+            swapper,
+            amountPcv,
+            sourceAsset,
+            destinationAsset
+        );
+
+        if (!ignoreBalanceChecks) {
+            /// TODO simplify this by passing delta of starting balance instead of calling balance again on each pcv deposit
+            /// record how balanced the system is before the PCV movement
+            int256 sourceVenueBalanceAfter = getVenueBalance(source, totalPcv);
+            int256 destinationVenueBalanceAfter = getVenueBalance(
+                destination,
+                totalPcv
+            );
+
+            /// source and dest venue balance measures the distance from being perfectly balanced
+
+            /// validate source venue balance became more balanced
+            _checkBalance(
+                sourceVenueBalance,
+                sourceVenueBalanceAfter,
+                "MarketGovernance: src more imbalanced"
+            );
+
+            /// validate destination venue balance became more balanced
+            _checkBalance(
+                destinationVenueBalance,
+                destinationVenueBalanceAfter,
+                "MarketGovernance: dest more imbalanced"
+            );
+        }
+    }
+
+    /// @notice helper function to validate balance moved in the right direction after a pcv movement
+    function _checkBalance(
+        int256 balanceBefore,
+        int256 balanceAfter,
+        string memory reason
+    ) private pure {
+        require(
+            balanceBefore < 0 /// if balance is under weight relative to vcon staked, ensure it doesn't go over balance
+                ? balanceAfter > balanceBefore && balanceAfter <= 0 /// if balance is over weight relative to vcon staked, ensure it doesn't go under balance
+                : balanceAfter < balanceBefore && balanceAfter >= 0,
+            reason
+        );
+    }
+
+    /// @notice VCON rewards could be negative if a user is at a loss
+    /// @param user to check rewards from
+    /// @param venue to check rewards in
+    function getPendingRewards(
+        address user,
+        address venue
+    ) public view returns (int256) {
+        int256 vconStartIndex = venueUserStartingVconProfit[venue][user]
+            .toInt256();
+        int256 vconCurrentIndex = venueLastRecordedSharePrice[venue].toInt256();
+
+        if (vconStartIndex == 0) {
+            return 0; /// no interest if user has not entered the market
+        }
+
+        int256 vconRewards = (vconCurrentIndex - vconStartIndex) *
+            venueUserDepositedVcon[venue][user].toInt256();
+
+        return vconRewards;
+    }
+
+    /// @notice returns profit in VCON a user has accrued
+    /// does not update how much VCON a user has staked to save on gas
+    /// that updating happens in the calling function
+    function _harvestRewards(
```

Comment:
rename this function to something more descriptive like `_updateUserProfitIndex`